### PR TITLE
Feature/49140 mtc admin view pupil history

### PR DESCRIPTION
**Pull request by jon-shipley**
New feature "View pupil history" for STA Admin role

### CODE REVIEW

File: admin/services/pupil-history-service/data-access/pupil-history.data.service.spec.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import moment from 'moment'
```
Comment (GuyHarwood) [inline admin/services/pupil-history-service/data-access/pupil-history.data.service.spec.ts:1]: is this first steps toward putting spec files alongside implementation?  
i definitely prefer the approach, maybe worth a conversation on how we could possibly migrate existing 🤔

File: admin/services/pupil-history-service/data-access/pupil-history.data.service.ts [hunk 2]
```diff
@@ -0,0 +1,236 @@
+import moment from 'moment';
```
Comment (GuyHarwood) [inline admin/services/pupil-history-service/data-access/pupil-history.data.service.ts:1]: ive also noticed how you are bundling feature logic into its own directory.
again, i prefer this over wholesale 'services' folder etc

File: admin/services/pupil-history-service/data-access/pupil-history.data.service.ts [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+  groupId: number | null,
+  currentCheckId: number | null,
+  checkComplete: boolean,
+  restartAvailable: boolean,
+  attendanceId: number | null,
+  foreNameAlias: string | null,
+  lastNameAlias: string | null
+}
+
+export interface ISchoolData {
```
Comment (GuyHarwood) [inline admin/services/pupil-history-service/data-access/pupil-history.data.service.ts:47]: this kind of thing has become fragmented in `tslib`, so i would try and use more specific names like `pupilHistorySchoolData` if possible.  However, there are definitely generic types/models that might be useful to use globally.  School is definitely one of them

File: admin/services/pupil-history-service/data-access/pupil-history.data.service.ts [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+
+export interface IPupilHistoryData {
+  pupil: IPupilData,
+  checks: ICheckData[]
+  school: ISchoolData
+  restarts: IRestartData[]
+}
+
+export class PupilHistoryDataService {
+  public static async getPupil(pupilUuid: string): Promise<IPupilData> {
```
Comment (GuyHarwood) [inline admin/services/pupil-history-service/data-access/pupil-history.data.service.ts:80]: why static? is this so they can be consumed easily in the existing javascript? or just a preference/design choice

In `tslib` we have interfaces for data services, and mocks complying to those interfaces are injected, which is nice and clean.  are you proposing a different way for `admin`?

File: admin/services/set-validation.service.js [hunk 5]
```diff
@@ -4,12 +4,12 @@ const setValidatorService = {
   /**
    * Utility function to validate that incoming form ids, match ids from objects in the second param
    * E.g. The first param is incoming form data from the UI, and the second are objects retrieved from the DB
-   * @param {Array.<number>} formIds
+   * @param {Array.<number>} a1
```
Comment (GuyHarwood) [inline admin/services/set-validation.service.js:7]: whats the purpose of a less descriptive parameter?

**Review by GuyHarwood — COMMENTED** [review]
Looking good!
few things to discuss around the typescript introduction.   great to see it happening :)

File: admin/services/pupil-history-service/data-access/pupil-history.data.service.ts (hunk 3 above)
Comment (jon-shipley) [reply admin/services/pupil-history-service/data-access/pupil-history.data.service.ts]: Good point. The naming has been clarified.

File: admin/routes/pupil-pin.js [hunk 6]
```diff
@@ -19,7 +19,6 @@ router.get(
   getSelectOfficialOrTryItOutPinGen
 )
 
-// TODO: JMS: Move new page to replace the overview page which has a lot of good behaviour
```
Comment (GuyHarwood) [inline admin/routes/pupil-pin.js:22]: is this a reminder for a further PR, or something to address in this one?

**Review by GuyHarwood — APPROVED** [review]
LGTM 💯

**Pull request merged**